src/Initializer/time_stepping/LtsLayout.cpp

```diff
@@ -59,34 +59,32 @@ seissol::initializers::time_stepping::LtsLayout::LtsLayout():
 
 seissol::initializers::time_stepping::LtsLayout::~LtsLayout() {
   // free memory of member variables
-  if( m_cellTimeStepWidths       != NULL ) delete[] m_cellTimeStepWidths;
-  if( m_cellClusterIds           != NULL ) delete[] m_cellClusterIds;
-  if( m_globalTimeStepWidths     != NULL ) delete[] m_globalTimeStepWidths;
-  if( m_globalTimeStepRates      != NULL ) delete[] m_globalTimeStepRates;
-  if( m_numberOfPlainGhostCells  != NULL ) delete[] m_numberOfPlainGhostCells;
-  if( m_plainGhostCellClusterIds != NULL ) for( unsigned int l_rank = 0; l_rank < m_plainNeighboringRanks.size(); l_rank++ ) delete[] m_plainGhostCellClusterIds[l_rank];
-  if( m_plainGhostCellClusterIds != NULL ) delete[] m_plainGhostCellClusterIds;
-  if( m_plainCopyRegions         != NULL ) delete[] m_plainCopyRegions;
+  delete[] m_cellTimeStepWidths;
+  delete[] m_cellClusterIds;
+  delete[] m_globalTimeStepWidths;
+  delete[] m_globalTimeStepRates;
+  delete[] m_numberOfPlainGhostCells;
+  if (m_plainGhostCellClusterIds != nullptr) {
+    for( unsigned int l_rank = 0; l_rank < m_plainNeighboringRanks.size(); l_rank++ ) {
+      delete[] m_plainGhostCellClusterIds[l_rank];
+    }
+  }
+  delete[] m_plainGhostCellClusterIds;
+  delete[] m_plainCopyRegions;
 }
 
 void seissol::initializers::time_stepping::LtsLayout::setMesh( const MeshReader &i_mesh ) {
   // TODO: remove the copy by a pointer once the mesh stays constant
-  std::vector<Element> const& elements = i_mesh.getElements();
-  for( unsigned int l_cell = 0; l_cell < elements.size(); l_cell++ ) {
-    m_cells.push_back( elements[l_cell] );
-  }
-  std::vector<Fault> const& fault = i_mesh.getFault();
-  for( unsigned int i = 0; i < fault.size(); ++i ) {
-    m_fault.push_back( fault[i] );
-  }
+  m_cells = i_mesh.getElements();
+  m_fault = i_mesh.getFault();
```

Comment:
Are you sure that copying is unnecessary? (I don't recall why it was in there in the first place...)

Reply:
Dunno. It works without copying - I remember that the mesh reader was freed quite early but currently it's freed only at the end of the simulation. Not sure what's the best way here...